data-structures-queues/take-next-smallest.js

```diff
@@ -0,0 +1,15 @@
+/* exported takeNextSmallest */
+
+// return the first value that is <= it's successor
+// enq all until it's found
+
+function takeNextSmallest(queue) {
+  let bucket;
+  while (true) {
```

Comment:
yehhh, i know. while-true is cheese. but it works, it does the Thing! :D

Reply:
Yes, 🤮 , LOL Maybe using something similar to what you used in the `if` statement as the condition for the `while` would work... 🤔